### PR TITLE
EN-4120: unit tests for rewards

### PR DESCRIPTION
**Pull request by bogdan-rosianu**

### CODE REVIEW

File: process/rewardTransaction/interceptedRewardTransaction_test.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
+package rewardTransaction
```
Comment (AdoAdoAdo) [inline process/rewardTransaction/interceptedRewardTransaction_test.go]: we should use rewardTransaction_test instead as package

File: process/rewardTransaction/interceptor_test.go [hunk 2]
```diff
@@ -0,0 +1,277 @@
+package rewardTransaction
```
Comment (AdoAdoAdo) [inline process/rewardTransaction/interceptor_test.go]: use rewardTransaction_test package instead

File: process/rewardTransaction/process_test.go [hunk 3]
```diff
@@ -0,0 +1,286 @@
+package rewardTransaction
```
Comment (AdoAdoAdo) [inline process/rewardTransaction/process_test.go]: rewardTransaction_test

File: process/rewardTransaction/interceptedRewardTransaction_test.go (hunk 1 above)
Comment (bogdan-rosianu) [reply process/rewardTransaction/interceptedRewardTransaction_test.go]: done

File: process/rewardTransaction/interceptor_test.go (hunk 2 above)
Comment (bogdan-rosianu) [reply process/rewardTransaction/interceptor_test.go]: done

File: process/rewardTransaction/process_test.go (hunk 3 above)
Comment (bogdan-rosianu) [reply process/rewardTransaction/process_test.go]: done

File: process/block/preprocess/rewardsHandler_test.go [hunk 4]
```diff
@@ -456,3 +459,118 @@ func TestRewardsHandler_GetAllCurrentFinishedTxs(t *testing.T) {
...
+		&mock.SpecialAddressHandlerMock{},
+		&mock.HasherMock{},
+		&mock.MarshalizerMock{},
+		mock.NewMultiShardsCoordinatorMock(3),
+		&mock.AddressConverterMock{},
+		&mock.ChainStorerMock{},
+		tdp.RewardTransactions(),
+	)
+
+	assert.Nil(t, th.protocolRewards)
```
Comment (miiu96) [inline process/block/preprocess/rewardsHandler_test.go]: add empty line

File: process/rewardTransaction/process_test.go [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/rewardTx"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/mock"
+	"github.com/ElrondNetwork/elrond-go/process/rewardTransaction"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewRewardTxProcessorNilAccountsDbShouldErr(t *testing.T) {
```
Comment (miiu96) [inline process/rewardTransaction/process_test.go]: where is _ in test name ?
maybe you should rename with "TestNewRewardTxProcessor_NilAccountsDbShouldErr".
If you change name you should do that for all tests from this file.

File: process/block/preprocess/rewardsHandler_test.go (hunk 4 above)
Comment (bogdan-rosianu) [reply process/block/preprocess/rewardsHandler_test.go]: done

File: process/rewardTransaction/process_test.go (hunk 5 above)
Comment (bogdan-rosianu) [reply process/rewardTransaction/process_test.go]: done

**Pull request merged**